### PR TITLE
Helm Insecure Arg Handling

### PR DESCRIPTION
**Pull request by dbw7**

### CODE REVIEW

File: pkg/helm/helm.go [hunk 1]
```diff
@@ -170,6 +178,13 @@ func pullCommand(chart string, repo *image.HelmRepository, version, destDir stri
 		args = append(args, "--destination", destDir)
 	}
 
+	if repo.SkipTLSVerify {
+		args = append(args, "--insecure-skip-tls-verify")
+	}
+	if repo.PlainHTTP {
+		args = append(args, "--plain-http")
+	}
```
Comment (atanasdinov) [inline pkg/helm/helm.go:186]: Aren't these two mutually exclusive? `else if` should be more appropriate to convey it.

File: pkg/helm/helm.go [hunk 2]
```diff
@@ -114,6 +118,10 @@ func registryLoginCommand(host string, repo *image.HelmRepository, output io.Wri
 		args = append(args, "--username", repo.Authentication.Username, "--password", repo.Authentication.Password)
 	}
 
+	if repo.SkipTLSVerify || repo.PlainHTTP {
```
Comment (atanasdinov) [inline pkg/helm/helm.go:121]: Is a `PlainHTTP` check appropriate here? Official docs state:
```
      --insecure           allow connections to TLS registry without certs
```

File: pkg/helm/helm_test.go [hunk 3]
```diff
@@ -127,6 +174,54 @@ func TestRegistryLoginCommand(t *testing.T) {
...
+			name: "Valid registry with auth and plain HTTP",
+			host: "registry-1.docker.io",
+			repo: &image.HelmRepository{
+				Name: "apache-repo",
+				URL:  "oci://registry-1.docker.io/bitnamicharts/apache",
+				Authentication: image.HelmAuthentication{
+					Username: "user",
+					Password: "pass",
+				},
+				SkipTLSVerify: true,
```
Comment (atanasdinov) [inline pkg/helm/helm_test.go:211]: This is duplicate of the previous test case.

File: pkg/helm/helm_test.go [hunk 4]
```diff
@@ -213,6 +308,80 @@ func TestPullCommand(t *testing.T) {
...
+					Username: "user",
+					Password: "pass",
+				},
+				PlainHTTP: true,
+			},
+			expectedArgs: []string{
+				"helm",
+				"pull",
+				"oci://registry-1.docker.io/bitnamicharts/apache",
+				"--plain-http",
```
Comment (atanasdinov) [inline pkg/helm/helm_test.go:382]: Pardon my ignorance - does this even make sense? Isn't the OCI protocol completely separate?

File: pkg/image/validation/kubernetes.go [hunk 5]
```diff
@@ -253,6 +276,36 @@ func validateRepo(repo *image.HelmRepository, seenHelmRepos map[string]bool) []F
...
+func validateHelmRepoArgs(repo *image.HelmRepository) []FailedValidation {
+	var failures []FailedValidation
+
+	if repo.SkipTLSVerify && repo.PlainHTTP {
+		failures = append(failures, FailedValidation{
+			UserMessage: fmt.Sprintf("Helm repository 'plainHTTP' and 'skipTLSVerify' fields for %q cannot both be true.", repo.Name),
+		})
+	}
+
+	if strings.HasPrefix(repo.URL, "http://") && !repo.PlainHTTP {
```
Comment (atanasdinov) [inline pkg/image/validation/kubernetes.go:288]: This and the checks below seem quite complex. I'd generally suggest to `url.Parse()`, use a switch statement on the `Scheme` field. Feel free to ignore the suggestion if it is not applicable.

File: pkg/helm/helm_test.go (hunk 4 above)
Comment (dbw7) [reply pkg/helm/helm_test.go:382]: I can’t say I fully understand it just yet but from what I understand at the moment is that it still uses HTTP(s) under the hood:
```
curl -u "user:root" "http://192.168.1.185:5000/v2/_catalog"
{"repositories":["apache/apache","rancher/rancher"]}
```

```
08:07:10 danny@ubuntu ~ → helm registry login http://192.168.1.185:5000
Username: user
Password:
INFO[0002] Error logging in to endpoint, trying next endpoint  error="Get \"https://192.168.1.185:5000/v2/\": http: server gave HTTP response to HTTPS client"
Error: Get "https://192.168.1.185:5000/v2/": http: server gave HTTP response to HTTPS client

08:07:27 danny@ubuntu ~ → helm registry login http://192.168.1.185:5000 --insecure
Username: user
Password:
WARN[0001] insecure registry http://192.168.1.185:5000 should not contain 'http://' and 'http://' has been removed from the insecure registry config
INFO[0001] Error logging in to endpoint, trying next endpoint  error="Get \"https://192.168.1.185:5000/v2/\": http: server gave HTTP response to HTTPS client"
Login Succeeded
```

File: pkg/helm/helm_test.go (hunk 3 above)
Comment (dbw7) [reply pkg/helm/helm_test.go:211]: Good catch, meant for that to be `plainHTTP: true`.

File: pkg/image/validation/kubernetes.go (hunk 5 above)
Comment (dbw7) [reply pkg/image/validation/kubernetes.go]: I don't think a switch statement would work here but I did switch over to using `url.Parse()`.

File: pkg/image/validation/kubernetes.go [hunk 6]
```diff
@@ -218,6 +222,25 @@ func validateChart(chart *image.HelmChart, imageConfigDir string) []FailedValida
 func validateRepo(repo *image.HelmRepository, seenHelmRepos map[string]bool) []FailedValidation {
 	var failures []FailedValidation
 
+	parsedURL, err := url.Parse(repo.URL)
+	if err != nil {
+		zap.S().Errorf("url '%s' could not be parsed: %s", repo.URL, err)
```
Comment (atanasdinov) [inline pkg/image/validation/kubernetes.go:227]: We need to return here - `parsedURL` is nil.

**Pull request merged**